backend/analytics_server/app.py

```diff
@@ -36,5 +37,19 @@
 configure_db_with_app(app)
 initialize_database(app)
 
+def handle_exception(e):
```

Comment:
@dominicghizzoni  I agree with @thesujai  here. The code is 50% there. I would suggest going through this doc: 
https://flask.palletsprojects.com/en/2.3.x/errorhandling/#registering, https://flask.palletsprojects.com/en/2.3.x/errorhandling/#returning-api-errors-as-json. 

After going through these and the example you would have an idea of what kind of a code.
I think [Generic Exception handlers](https://flask.palletsprojects.com/en/2.3.x/errorhandling/#generic-exception-handlers) can be applied in our case